src/common/ownsql.cpp

```diff
@@ -262,18 +271,26 @@ int SqlQuery::prepare(const QString &sql, bool allow_failure)
             qCWarning(lcSql) << "Sqlite prepare statement error:" << _error << "in" << _sql;
             ENFORCE(allow_failure, "SQLITE Prepare error");
         }
+        _sqldb->_queries.insert(this);
     }
     return _errId;
 }
 
+// There is no overloads to QByteArray::startWith that takes Qt::CaseInsensitive
+template <int N>
+static bool startsWithInsensitive(const QByteArray &a, const char (&cal)[N])
```

Comment:
Mmh, is this complication for the sake of optimization really worth it?

Reply:
Where is the complication? This function looks simple to me. How would you simplify it? Would you think it is simpler if it was taking a const char* and using strlen?

Reply:
Maybe it's just me being unused to passing arrays this way. I think strlen use would have surprised me less. Taking a `QByteArray` and using `QByteArrayLiteral` might be the most straightforward path?